pkg/cloud/openstack/services/networking/loadbalancer.go

```diff
@@ -0,0 +1,437 @@
+package networking
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/floatingips"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/lbaas_v2/listeners"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/lbaas_v2/loadbalancers"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/lbaas_v2/monitors"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/lbaas_v2/pools"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	providerv1 "sigs.k8s.io/cluster-api-provider-openstack/pkg/apis/openstackproviderconfig/v1alpha1"
+	constants "sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/openstack/contants"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/util"
+	"time"
+)
+
+func (s *Service) ReconcileLoadBalancer(clusterName string, clusterProviderSpec *providerv1.OpenstackClusterProviderSpec, clusterProviderStatus *providerv1.OpenstackClusterProviderStatus) error {
+
+	if clusterProviderSpec.ExternalNetworkID == "" {
+		klog.V(3).Info("No need to create loadbalancer, due to missing ExternalNetworkID")
+		return nil
+	}
+	if clusterProviderSpec.APIServerLoadBalancerFloatingIP == "" {
```

Comment:
The comment about floating IP getting created if not provided is false then?

Reply:
Oh wait, is it because the floating IP field would start out as nil, so if it's empty then something else didn't go right when setting it?

Reply:
The idea is: You configure that the APIServerLoadBalancer should be managed. Only in this case this method here is used. Then it's also required that the FloatingIP is set.